### ISSUE TITLE
Not working with Angular 11

### ISSUE THREAD

**Issue by Ansuken** [KevCJones/ngx-simple-modal#73]
I use Angular 11.0.9 and "ngx-simple-modal": "^1.4.14" but it doesn't work.

Error: node_modules/ngx-simple-modal/simple-modal/simple-modal.module.d.ts:5:97 - error TS2314: Generic type 'ModuleWithProviders<T>' requires 1 type argument(s).

Are there any plans to update the lib soon?

Thanks!!

**Comment by KevCJones** [comment]
I see, OK np I will take a look later and setup an angular 11 harness and
see what breaking change its been hit with

**Comment by KevCJones** [comment]
Hi @Ansuken 

I've not been able to reproduce your reported issue. 

I've bumped my version since and added an ng11 demo folder in the project, and also created this https://stackblitz.com/edit/ngx-simple-modal-ng11?devtoolsheight=33&file=src/index.html so you can see it in action and maybe see what you have that i do not. 

Cheers

Kev

**Comment by Ansuken** [comment]
Hi!

Maybe the problem is when you upgrade from older Angular version to 11. 

I fix this issue adding this at the end of app.module.ts

```
declare module "@angular/core" {
  interface ModuleWithProviders<T = any> {
    ngModule: Type<T>;
    providers?: Provider[];
  }
}
```

When you open the file in the Error: 
`Error: node_modules/ngx-simple-modal/simple-modal/simple-modal.module.d.ts:5:97 - error TS2314: Generic type 'ModuleWithProviders<T>' requires 1 type argument(s).`

This is the code:
```
import { ModuleWithProviders } from '@angular/core';
import { SimpleModalServiceConfig } from './simple-modal.service';
import { SimpleModalOptions } from './simple-modal-options';
export declare class SimpleModalModule {
    static forRoot(config: SimpleModalServiceConfig, defaultModalOptions?: SimpleModalOptions): ModuleWithProviders;
    constructor();
}
```

There is an error in ModuleWithProviders that says: 'A wrapper around an NgModule that associates it with providers. Usage without a generic type is deprecated.'

In your example I see that there is no problem, works fine, so I don't understand how doing a fresh install of the ngx-simple-modal package can give me that error.

PS: Sorry about my English :)

Thanks for your time.

**Comment by KevCJones** [comment]
That is interesting - I think this is a typings issue possibly then. I can easily specify the typing and I might do that anyway but when I checked my project I explored the core.d.ts file and 

```
**
 * A wrapper around an NgModule that associates it with the providers.
 *
 * @param T the module type. In Ivy applications, this must be explicitly
 * provided.
 *
 * @publicApi
 */
export declare interface ModuleWithProviders<T = any /** TODO(alxhub): remove default when callers pass explicit type param */> {
    ngModule: Type<T>;
    providers?: Provider[];
}
```
This made me think and I've gone to look at the repo and sure enough 

https://github.com/angular/core-builds/blob/11.1.x/core.d.ts#L4253

This is probably very much a problem that is going to happen more and more as people pull the latest core typings.  Following that annotation URL hint brings me here.

https://angular.io/guide/deprecations#modulewithproviders-type-without-a-generic

So in short, you are right, there is an issue 👍 I am assuming that the demo site (stackblitz) and my local version resolved an older version of the tools. I'm still not sure why it works locally either because checking my ng11 demo folder, the core.d.ts also has no =any but it let me fly. All I can assume is that there was some implicit `any` and now there is not.

I will sort out a quick update and bump now.  

Thanks for the contribution and your English is perfect - don't sweat it

**Comment by KevCJones** [comment]
Update to version #1.4.16 and you shouldn't need your shim

**Comment by Ansuken** [comment]
Works correctly!!

I really appreciate your effort.

Thanks!
